### ISSUE TITLE
Crash when deleting fitting

### ISSUE THREAD

**Issue by twothe** [pyfa-org/Pyfa#1330]
I just deleted a fitting (with an open tab of that fitting), which caused a crash.
I was not using any projections on or with that ship, but I did use projection in a different fitting of a different ship earlier (that fitting was neither deleted nor open nor was any projection still active).

```
OS version: 	Windows-8-6.2.9200
Python: 	2.7.10
wxPython: 	3.0.2.0
SQLAlchemy: 	1.0.5
Logbook: 	1.0.0
pyfa version: 1.33.1 Stable - Lifeblood 1.0
pyfa root: D:\Games\pyfa
save path: C:\Users\Two\.pyfa
fs encoding: mbcs

EXCEPTION: Class '__builtin__.NoneType' is not mapped

  File "D:\Games\pyfa\library.zip\gui\builtinShipBrowser\sfBrowserItem.py", line 333, in OnLeftUp
    btn.DoCallback()
  File "D:\Games\pyfa\library.zip\gui\builtinShipBrowser\sfBrowserItem.py", line 50, in DoCallback
    self.callback()
  File "D:\Games\pyfa\library.zip\gui\builtinShipBrowser\fitItem.py", line 351, in deleteBtnCB
    self.deleteFit()
  File "D:\Games\pyfa\library.zip\gui\builtinShipBrowser\fitItem.py", line 368, in deleteFit
    sFit.deleteFit(self.fitID)
  File "D:\Games\pyfa\library.zip\service\fit.py", line 182, in deleteFit
    if projection.victim_fit != fit and projection.victim_fit in eos.db.saveddata_session:  # GH issue #359
  File "C:\python-2.7.10\lib\site-packages\sqlalchemy\orm\session.py", line 1956, in __contains__

```

**Comment by blitzmann** [comment]
So, there's been delete issues for a long while now. This particular bug seems related to an attempted fix I tried to do a while back.

Unfortunately these things are extremely hard to track down or reproduce. I've been needing to do an overhaul of the delete procedure for a while, but I have a couple other projects right now (that may open up some doors on how to address things like this), and while this is annoying to have pop up, I don't believe it causes any negative effects. Will keep this open for a time when i can delve back into it. :)

Would you mind attaching your `%userprofile%/.pyfa/` that contains this error? There might be some information in it that might help as well as to how it may have happened.

**Comment by JSinx** [comment]
I always see this error when delete last drone from stack

**Comment by twothe** [comment]
I attached the pyfa user folder with the relevant log file:

[pyfa.zip](https://github.com/pyfa-org/Pyfa/files/1433411/pyfa.zip)

**Comment by blitzmann** [comment]
@twothe thanks for the files. Removed the download link from the comment and uploaded just the log, as the `saveddata.db` file may contain your API keys / secret AT fits / other crap that you may not want floating around ;)

[Pyfa-2017-10-31.log](https://github.com/pyfa-org/Pyfa/files/1434952/Pyfa-2017-10-31.log)

**Comment by blitzmann** [comment]
@JSinx Is the error that you get always reproducible? Can you post the error text here? doesn't sound like it's the same issue :/

**Comment by taleden** [comment]
I've seen this crash pretty frequently while cleaning up duplicate fittings. I recently came back to the game and imported all my old EFT and Pyfa fittings into the latest Pyfa; since I previously kept the two somewhat in sync, this left me with many duplicates (two fits for the same hull with the same name).

When I open two such fits, confirm that they have all the same modules and then delete one of them (while both are still among the opened fit tabs and one is the currently active fit), I sometimes (~25-50%?) get a crash with "pyfa.exe has stopped working" but no error or stack trace. I also tried running from source with the "-d" option and then the error is instead "python.exe has stopped working", but still no message or stack trace on screen or in the console log.
